src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1155,6 +1155,7 @@ void op_context::for_each_fetch_partition(Func&& f) const {
                 .topic_partition = {p.topic->name, part.partition_index},
                 .max_bytes = part.max_bytes,
                 .fetch_offset = part.fetch_offset,
+                .current_leader_epoch = part.current_leader_epoch,
```

Comment:
We didn't pass `current_leader_epoch` at all for the first fetch in the session (e.g. after broker restart) or if session's weren't used at all so we never checked it.

> the epoch check is bypassed because current_leader_epoch defaults to no leader epoch, is that correct?

Yes.

> what happens to the subsequent OffsetForLeaderEpoch() request? Is that made with the right epoch?

In this case, yes. If we have messages with epoch > than current_leader_epoch then fencing must happen. The franz-go behavior we have seen is not because it does internally some epoch incrementation but because it have already seen messages from newer epoch so it is using the epoch from these messages to make subsequent requests.